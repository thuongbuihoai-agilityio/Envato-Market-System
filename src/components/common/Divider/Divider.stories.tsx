import { StoryObj, Meta } from '@storybook/react';

// Components
import Divider from '@components/common/Divider';

const meta: Meta<typeof Divider> = {
  title: 'Custom Components/Divider',
  component: Divider,
  argTypes: {
    content: {
      description: 'Content of the divider text field',
    },
  },
};

type Story = StoryObj<typeof Divider>;

export const Default: Story = {
  args: {
    content: 'Divider Content',
  },
};

export default meta;
