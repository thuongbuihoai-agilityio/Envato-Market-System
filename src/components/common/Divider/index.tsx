import {
  AbsoluteCenter,
  Box,
  Divider as ChakraProvider,
} from '@chakra-ui/react';

type TDividerProps = {
  content: string;
};

const Divider = ({ content }: TDividerProps) => (
  <Box position="relative" w="full" m="auto" py="10">
    <ChakraProvider w="full" bg="gray.500" />

    <AbsoluteCenter
      _light={{ bg: '#fff' }}
      _dark={{ bg: 'secondary.600', height: '1px' }}
      px="4"
      color="gray.500"
      top="50%"
      transform="translate(-50%, -50%)"
    >
      {content}
    </AbsoluteCenter>
  </Box>
);

export default Divider;
