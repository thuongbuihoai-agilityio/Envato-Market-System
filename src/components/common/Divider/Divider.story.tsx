import { StoryObj, Meta } from '@storybook/react';

// Components
import Divider from '@components/common/Divider';

const meta: Meta<typeof Divider> = {
  component: Divider,
};

type Story = StoryObj<typeof Divider>;

export const Default: Story = {
  args: {
    content: 'Divider Content',
  },
};

export default meta;
