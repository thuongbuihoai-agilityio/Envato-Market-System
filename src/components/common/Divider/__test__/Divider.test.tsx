import { render } from '@testing-library/react';

// Components
import Divider from '@components/common/Divider';

test('renders DividerCustom with content', () => {
  const { container } = render(<Divider content="Test Content" />);

  expect(container).toMatchSnapshot();
});
