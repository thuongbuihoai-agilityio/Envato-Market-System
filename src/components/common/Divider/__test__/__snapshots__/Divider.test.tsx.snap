// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DividerCustom with content 1`] = `
<div>
  <div
    class="css-1wqp5dg"
  >
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-ri09he"
    />
    <div
      class="css-1u70y5c"
    >
      Test Content
    </div>
  </div>
</div>
`;
